src/evaluations/evaluator.py

```diff
@@ -14,14 +14,16 @@
 """Evaluator for cardinality estimators."""
 
 import copy
+import itertools
 import os
 import pickle
 import shutil
 import time
 
 from absl import logging
-
 import numpy as np
+from pathos.multiprocessing import ProcessPool, cpu_count
```

Comment:
Why pathos and not the native multiprocessing library?

Reply:
This is explained in the PR summary, but basically we need to serialize the arguments we are inputting into the process we are spawning when using multiprocessing. In the multiprocessing library, the arguments are serialized with Pickle which cannot handle some of the arguments we are inputting (namely, local functions inside of our estimators configs). Pathos is a library which has all the functionality of Python's native multiprocessing library, but uses Dill to serialize arguments instead, allowing it to serialize all of our arguments with the least amount of code change to our project.

Reply:
It also looks, Craig, like Pathos might allow us to easily wield an army of multiple machines on different clouds but I'm not sure yet:
https://github.com/uqfoundation/pathos